### ISSUE TITLE
Optimise purging without losing too many genes

### ISSUE THREAD

**Issue by rotoke** [dfguan/purge_dups#42]
Dear Dengfeng,

I am using purge_dups on a diploid canu assembly (haplotypes not collapsed during assembly because of high heterozygosity). I have run step 1-3 of the pipeline (I don't have an alternative assembly, so no step 4), but unfortunately, purging is incomplete and leads to gene loss. I would be happy to get some advice what I could adjust.

_Assembly stats_

```
                                   Assumed genome size (Mbp)    2300.00 (= diploid genome size)

                                         Number of scaffolds      38248
                                     Total size of scaffolds 2205269562
  Total scaffold length as percentage of assumed genome size      95.9%
             Useful amount of scaffold sequences (>= 25K nt) 1962847776
                        % of estimated genome that is useful      85.3%
                                            Longest scaffold    7957226
                                           Shortest scaffold       1152
                                 Number of scaffolds > 1K nt      38248 100.0%
                                Number of scaffolds > 10K nt      23069  60.3%
                               Number of scaffolds > 100K nt       4412  11.5%
                                 Number of scaffolds > 1M nt        192   0.5%
                                Number of scaffolds > 10M nt          0   0.0%
                                          Mean scaffold size      57657
                                        Median scaffold size      13379
                                         N50 scaffold length     249133
                                          L50 scaffold count       1931
                                        NG50 scaffold length     228965
                                         LG50 scaffold count       2129
```

_BUSCO stats before purging_

```
       C:97.0%[S:20.8%,D:76.2%],F:0.6%,M:2.4%,n:2326
        2257    Complete BUSCOs (C)
        484     Complete and single-copy BUSCOs (S)
        1773    Complete and duplicated BUSCOs (D)
        14	Fragmented BUSCOs (F)
        55	Missing BUSCOs (M)
        2326    Total BUSCO groups searched  
```

_Automatic threshold selection_

![Cutoffs_purgedups_PB](https://user-images.githubusercontent.com/67823452/87336178-59e1cf00-c539-11ea-9056-827e6cdee8cc.png)

_BUSCO stats after purging_

```
        C:95.3%[S:61.7%,D:33.6%],F:0.8%,M:3.9%,n:2326
        2216    Complete BUSCOs (C)
        1434    Complete and single-copy BUSCOs (S)
        782     Complete and duplicated BUSCOs (D)
        19	Fragmented BUSCOs (F)
        91	Missing BUSCOs (M)
        2326    Total BUSCO groups searched
```

In summary, the purging reduces duplication from ca. 76% to 33% (which seems still relatively high to me), but I'm also losing ca. 2% of genes. I have tried different sets of manual thresholds, which all resulted in better purging, but also bigger gene loss. So maybe I'm not setting these right.

- Do you have any recommendations on the thresholds to improve the purging without losing too many genes?

- Is there any other parameter I could tweak to improve the purging? (e.g. increasing the minimum alignment score as in #30?)

Thanks a lot for your help!

Best,
Roman

**Comment by rotoke** [comment]
Update: I also have some Illumina short reads, which I mapped to the assembly:

![Cutoffs_purgedups_illumina](https://user-images.githubusercontent.com/67823452/87786999-d541ca00-c832-11ea-85dc-5dc43928d1d3.png)

Using the suggested thresholds improves the purging quite a bit, but 16% duplication still remains:

```
        C:95.0%[S:79.0%,D:16.0%],F:0.9%,M:4.1%,n:2326
        2209    Complete BUSCOs (C)
        1838    Complete and single-copy BUSCOs (S)
        371     Complete and duplicated BUSCOs (D)
        20	Fragmented BUSCOs (F)
        97	Missing BUSCOs (M)
        2326    Total BUSCO groups searched
```

 I assume there is a tradeoff between loss of duplication and loss of genes, so the trick is to get the settings right to end up with an optimal balance.
